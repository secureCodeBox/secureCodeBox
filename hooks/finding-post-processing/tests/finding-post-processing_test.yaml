suite: Full Snapshot
tests:
  - it: matches the snapshot
    chart:
      version: 0.0.0
      appVersion: 0.0.0        
    set:
      ## Finding Post Processing Hook Specific Configuration
      rules:
        [
          {
            matches:
              [
                {
                  anyOf:
                    [
                      {
                        category: "Open Port",
                        attributes: {port: 21, state: open},
                      },
                      {
                        category: "Open Port",
                        attributes: {port: 389, state: open},
                      },
                    ],
                },
              ],
            override: {severity: "high", description: "Telnet is bad"},
          },
        ]
      ## Generic Configuration        
      imagePullSecrets: [{name: foo}]
      hook:
        labels: {foo: bar}
        annotations: {foo: bar}
        priority: 100
        ttlSecondsAfterFinished: 100
        tolerations:
          [{key: foo, operator: Exists, value: "value1", effect: NoSchedule}]
        affinity:
          nodeAffinity:
            requiredDuringSchedulingIgnoredDuringExecution:
              nodeSelectorTerms:
                - matchExpressions:
                    - {key: foo, operator: In, values: [bar, baz]}
        resources: {requests: {cpu: "100m", memory: "100Mi"}}
        extraVolumes: [{name: foo, emptyDir: {}}]
        extraVolumeMounts: [{name: foo, mountPath: /foo}]
        env: [{name: foo, value: bar}]
    asserts:
      - matchSnapshot: {}
