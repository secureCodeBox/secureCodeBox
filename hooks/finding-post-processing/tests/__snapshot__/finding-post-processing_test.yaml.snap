matches the snapshot:
  1: |
    raw: |2

      FindingPostProcessing Hook deployed.
      This will add postprocessing on every finding in this namespace matching these rules: [{"matches":[{"anyOf":[{"attributes":{"port":21,"state":"open"},"category":"Open Port"},{"attributes":{"port":389,"state":"open"},"category":"Open Port"}]}],"override":{"description":"Telnet is bad","severity":"high"}}].
  2: |
    apiVersion: execution.securecodebox.io/v1
    kind: ScanCompletionHook
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: finding-post-processing
        app.kubernetes.io/version: 0.0.0
        foo: bar
        helm.sh/chart: finding-post-processing-0.0.0
      name: RELEASE-NAME-finding-post-processing
    spec:
      affinity:
        nodeAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
            nodeSelectorTerms:
              - matchExpressions:
                  - key: foo
                    operator: In
                    values:
                      - bar
                      - baz
      env:
        - name: RULES
          value: '[{"matches":[{"anyOf":[{"attributes":{"port":21,"state":"open"},"category":"Open Port"},{"attributes":{"port":389,"state":"open"},"category":"Open Port"}]}],"override":{"description":"Telnet is bad","severity":"high"}}]'
        - name: foo
          value: bar
      image: docker.io/securecodebox/hook-finding-post-processing:0.0.0
      imagePullSecrets:
        - name: foo
      priority: 100
      resources:
        requests:
          cpu: 100m
          memory: 100Mi
      tolerations:
        - effect: NoSchedule
          key: foo
          operator: Exists
          value: value1
      ttlSecondsAfterFinished: 100
      type: ReadAndWrite
      volumeMounts:
        - mountPath: /foo
          name: foo
      volumes:
        - emptyDir: {}
          name: foo
